src/gfx_layout_icu.cpp

```diff
@@ -243,7 +244,7 @@ int ICUParagraphLayout::ICULine::GetLeading() const
  */
 int ICUParagraphLayout::ICULine::GetWidth() const
 {
-	int length = 0;
+	float length = 0;
 	for (const auto &run : *this) {
 		length += run.GetAdvance();
 	}
```

Comment:
`return length;` needs to be `return ceil(length);` because our string system only deals with whole-units. You can verify this by looking at the effect on drop-down lists.